coordinator/tributary/src/block.rs

```diff
@@ -146,12 +152,14 @@ impl<T: Transaction> Block<T> {
     self.header.hash()
   }
 
-  pub(crate) fn verify(
+  pub(crate) fn verify<N: Network>(
     &self,
     genesis: [u8; 32],
     last_block: [u8; 32],
     mut locally_provided: HashMap<&'static str, VecDeque<T>>,
```

Comment:
We also need to verify unsigned TXs aren't being included multiple times.

Preferably, we enforce an order of Provided, Unsigned, Signed as well.